salt/modules/file.py

```diff
@@ -2093,14 +2093,10 @@ def replace(path,
                         # Content was found, so set found.
                         found = True
 
-                # Keep track of show_changes here, in case the file isn't
-                # modified
-                if show_changes or append_if_not_found or \
-                   prepend_if_not_found:
-                    orig_file = r_data.read(filesize).splitlines(True) \
-                        if isinstance(r_data, mmap.mmap) \
-                        else r_data.splitlines(True)
-                    new_file = result.splitlines(True)
+                orig_file = r_data.read(filesize).splitlines(True) \
```

Comment:
Sorry for the silence. Been away from my computer lately and sorry and I still haven't looked at this yet - no changes made so far.

As lorengorden pointed out, the solution can likely be optimised to reduce memory usage by not storing both files entirely in RAM to do a comparison. I'm happy for the changes to get merged as are for now and then later improved upon (going on the assumption that this module is almost always used for small files such as configuration files), but I don't mind holding off for a more memory-efficient implementation either if that's preferred.